x/orders/internal/keeper/keeper.go

```diff
@@ -41,15 +41,26 @@ func NewKeeper(storeKey sdk.StoreKey, cdc *codec.Codec, bk bank.Keeper, sk suppl
 func (k Keeper) PostOrder(
 	ctx sdk.Context,
 	owner sdk.AccAddress,
-	marketID dnTypes.ID,
+	assetCode types.AssetCode,
 	direction types.Direction,
 	price sdk.Uint,
 	quantity sdk.Uint,
 	ttlInSec uint64) (types.Order, error) {
 
-	market, err := k.marketKeeper.GetExtended(ctx, marketID)
+	marketsList := k.marketKeeper.GetList(ctx)
+
+	var market markets.MarketExtended
+	err := sdkErrors.Wrap(types.ErrWrongAssetCode, "not found")
+
+	for _, marketItem := range marketsList {
```

Comment:
I think we should extend the 'GetListFiltered' for orders.Keeper to support filtering by asset_code. That would make this part shorter

Reply:
Yes, fixed it